include/cadmium/core/modeling/coupled.hpp

```diff
@@ -32,13 +32,15 @@
 #include "port.hpp"
 #include "../exception.hpp"
 
+#include <iostream>
+
 namespace cadmium {
 	//! Couplings are just tuples <portFrom, portTo>
-    using coupling = std::tuple<const std::shared_ptr<PortInterface>, const std::shared_ptr<PortInterface>>;
+	using coupling = std::tuple<std::shared_ptr<PortInterface>, std::shared_ptr<PortInterface>>;
```

Comment:
Why do you need this mutable?

Reply:
You need it mutable to delete elements on couplings vectors. Otherwise gives error when deleting.